accepted/2024/[TS]PMUI-Readme-rendering.md

```diff
@@ -0,0 +1,213 @@
+
+Update `IPackageSearchMetadata` to include the `ReadmeFileUrl` field. 
+`LocalPackageSearchMetadata` will populate the field with the location of the README, if available, on the local disk.
+When a server implements the new `RegistrationsBaseUrl` resource type, then the `ReadmeFileUrl` field will be returned from the server if a README is available and be deserialized from the response.
+When a server implements the `ReadmeUriTemplate` resource type and `ReadmeFileUrl` is empty, it will be populated with the url to the README. 
+
+#### Downloading the README
+
+Update the `INuGetPackageFileService` to add a new method for downloading the README. 
+
+## Drawbacks
+
+MarkdownPreview control currently marked as obsolete since the interface has not been finalized.
+So when an upgrade is made we may have to change how we use the control.
+
+WebView2 controls always render ontop of other controls in the view.
+[Secnario 25254665](https://microsoft.visualstudio.com/Edge/_workitems/edit/25254665).
+PM UI needs to be updated to ensure items don't scroll off screen.
+
+`INuGetPackageFileService` in it's current implementation cannot download files from feeds that require authentication.
+
+## Rationale and alternatives
+By using an existing control we maintain consistency throughout the IDE and can rely on the owner to fix any bugs with the control.
+
+Due to concerns about performance we will not be downloading the full nupkg temporarily just to access the README.
+
+## Prior Art
+The IMarkdownPreview is currently being used when creating a new pull request inside of Visual Studio.
+![Alt text](../../meta/resources/READMEPMUI/PullRequestExperience.png) 
+
+nuget.org currently renders the README and our users will expect them to look the same.
+Ex. https://www.nuget.org/packages/Newtonsoft.Json#README-body-tab
+
+## Unresolved Questions
+1. Will we show README in the updates tab?
+    - Yes
+1. What do we show if the package has a README.txt instead of md?
+    - The README is spec is written as only accepting MD.
+    So we will use the nuspec to determine where the README is and treat it as md, even if the file is actually txt.
+1. What do we show if there is no README defined?
+    - Show the README tab with a message saying there is no README for the selected package/version.
+1. Where do we get the README from when it's not on the disk?
+    - Will document two ways for servers to provide the readme.
+    Updating the package metadata to include a link to the readme, and a direct download similar to the way nuspec and nupkg are implemented.
+1. Where are the README files saved in a package?
+    - The nuspec defines where to locate the README.
+1. What do we want the UX to be when an exception or error occurs while reading a README file?
+    - Inform the users that an exception occurred. If time permits allow for a retry.
+1. Do we want the README to update whenever a new version is selected for the current package?
+   - Yes.
```

Comment:
This might be a personal preference, but since these "unresolved questions" have answers now, I'd prefer the answers to be integrated throughout the rest of the document and removed from here.  Honestly, I'm not a fan of this "unresolved questions" section.

Reply:
Agreed.